webpack/es-bundle-core.babel.js

```diff
@@ -29,9 +29,10 @@ const result = configBuilder(
       ],
     },
     output: {
-      globalObject: "this",
-      library: "SwaggerUIBundle",
+      // with webpack 5 and commonjs2, there should not be a library[String] or library[Object]
+      // https://github.com/webpack/webpack/issues/12675
       libraryTarget: "commonjs2",
```

Comment:
In this PR description you mentioned:

> Note, library module now maps to swagger-ui-es-bundle-core instead of swagger-ui-es-bundle, which means that dependencies are no longer bundled as part of the npm package. This change should help enable better tree-shaking, as well as avoiding errors about multiple React versions that use React hooks [(Ref React Hook error)](https://reactjs.org/link/invalid-hook-call).

This is unfortunately not true. You can verify that by building the bundle without minification and mangling. All production dependencies are bundled in the bundle and webpack wraps them into it's special modules. Using tree-shaking is not possible. But that's basically how this bundle worked before and I always had problem understanding why this bundle name contains `es` in its name if it contains all the bundled prod deps and uses CommonJS module system. On top of that it's basically copy of `swagger-ui-es-bundle.js`. What what the motivation of using `commonjs2`? The only difference between `commonjs` and `commonjs2` module format is described here: https://github.com/webpack/webpack/issues/1114#issuecomment-105509929

In order to produce pure ESM bundle with possible tree-shaking and no prod deps bundled in the webpack config would need to look like this: https://github.com/char0n/create-react-app/blob/main/packages/react-scripts/config/webpack.config.bundle.esm.js

> Similarly, swagger-ui-react library module now includes swagger-ui-es-bundle-core instead of swagger-ui-es-bundle. This change should help alleviate issues when running swagger-ui-react with create-react-app@5, which was complaining of multiple react versions being loaded.

I haven't been able to reproduce this error.


The same that I've elaborated here applies for https://github.com/swagger-api/swagger-ui/blob/2e3395127d1e3a41931a4cd8b11dd4fe9411dd76/webpack/es-bundle.babel.js

Reply:
atm, minimizing changes to make work with webpack@5 and webpack-dev-server@4. I'm definitely up for changing and updating our bundles and/or pure ESM, but as discussed, would be breaking changes.

Reply:
> atm, minimizing changes to make work with webpack@5 and webpack-dev-server@4. I'm definitely up for changing and updating our bundles and/or pure ESM in a future PR, but as discussed, would be breaking changes.

Right, as I mentioned before overlooking difference between `es` and `es-core` was my mistake. `es-core` fragments doesn't allow tree-shaking but allow 3rd party bundlers to provide overrides, aliases and fallbacks, which is far better then using `swagger-ui.js`. Yes as we discussed we agreed on limiting changes to fragments to minimum, we'll introduce true ESM build fragment progressively later.

I'm still puzzled on `commonjs2` though. Can you elaborate more on this please?